### DISCUSSION TITLE
Error running update for the first time

### DISCUSSION THREAD

**Discussion by armzzz** [Kalanyr/gogrepo#78] [General]
Hi, I've just grabbed this script since GOG's current issues with their update tab is annoying. I've got Python 3.11 installed on Windows 11 along with the listed dependencies gogrepoc requires. I logged in successfully using 
`py gogrepoc.py login`
but when I try to run 
`py gogrepoc.py update -os windows -lang en`
I get the following error:

> 16:19:14 | loading local manifest...
> 16:19:14 | fatal...
> Traceback (most recent call last):
>   File "C:\Users\armaa\Downloads\_Temp\_GOG\gogrepoc.py", line 2891, in <module>
>     main(process_argv(sys.argv))
>   File "C:\Users\armaa\Downloads\_Temp\_GOG\gogrepoc.py", line 2630, in main
>     cmd_update(args.os, args.lang, args.skipknown, args.updateonly, not args.full, args.ids, args.skipids,args.skiphidden,args.installers,args.resumemode,args.strictverify)
>   File "C:\Users\armaa\Downloads\_Temp\_GOG\gogrepoc.py", line 1234, in cmd_update
>     gamesdb = load_manifest()
>               ^^^^^^^^^^^^^^^
>   File "C:\Users\armaa\Downloads\_Temp\_GOG\gogrepoc.py", line 348, in load_manifest
>     with codecs.open(filepath, 'rU', 'utf-8') as r:
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "< frozen codecs>", line 905, in open
> ValueError: invalid mode: 'rUb'

Any ideas what I'm doing wrong?

**Comment by Kalanyr** [comment]
I think you're trying to run it from inside the zip file which means the
state is inconsistent. You should extract it somewhere first.

**Reply by armzzz** [reply]
It's not in a zip file. 

I just noticed that the log message I pasted removed the file name, from the second to last line; it should say 
> File "< frozen codecs > ", line 905, in open

I kind of wonder if it's trying to actually open the local manifests file, which doesn't exist, so it's throwing this error.

**Comment by Kalanyr** [comment]
Hmm. I haven't used Windows 11 or Python 3.11 before but I'm moving to a
new window 11 computer now so I'll be able to troubleshoot this shortly.
The error doesn't really make sense because it's not being opened in the
mode claimed.

**Comment by Kalanyr** [comment]
Okay, I've looked into this. U is the default behavior for open (since
Python 3) and was removed as a valid flag in Python 3.11.

I'll have to issue a compatibility update.  Unfortunately I probably won't
get too in the next week or two. I'd recommend using Python 3.10 until I
get it sorted.

**Comment by armzzz** [comment]
Sweet. I'll install 3.10 for now. Thanks for looking into it.
